File: fixtures/scale/python/large/cipherlab/checks/check_pulsestream.py
from cipherlab.ciphers.pulsestream import pulsestream_encrypt, pulsestream_decrypt
from cipherlab.specs.pulsestream_spec import pulsestream_spec
from cipherlab.corpus.corpus_stream import corpus_stream


def check_pulsestream():
    spec = pulsestream_spec()
    for text in corpus_stream():
        sealed = pulsestream_encrypt(text, spec["key"])
        opened = pulsestream_decrypt(sealed, spec["key"])
        if opened != text:
            return False
    return spec["category"] == "stream"

File: fixtures/scale/ruby/large/lib/check_pulsestream.rb
require_relative 'pulsestream_encrypt'
require_relative 'pulsestream_decrypt'
require_relative 'corpus_stream'

def check_pulsestream
  corpus_stream.each do |text|
    sealed = pulsestream_encrypt(text, 10)
    opened = pulsestream_decrypt(sealed, 10)
    return false if opened != text
  end
  true
end
